src/pd/pd.rs

```diff
@@ -226,37 +335,37 @@ impl<T: PdClient> Runner<T> {
         handle.spawn(f)
     }
 
-    fn handle_heartbeat(
-        &self,
-        handle: &Handle,
-        region: metapb::Region,
-        peer: metapb::Peer,
-        region_stat: RegionStat,
-    ) {
+    fn handle_heartbeat(&mut self, handle: &Handle, region_id: u64, hb_task: Task) {
+        let cache = self.heartbeat_cache
+            .entry(region_id)
+            .or_insert_with(RegionHeartbeatCache::default);
+        info!("debug: before merge {:?}\n{:?}", cache, hb_task);
+        cache.merge_heartbeat(&self.db, hb_task);
+        info!("debug: after merge {:?}", cache);
+
         self.store_stat
             .region_bytes_written
-            .observe(region_stat.written_bytes as f64);
+            .observe(cache.written_bytes_delta as f64);
```

Comment:
Does it mean that the previous statistics are wrong?

Reply:
I think we are both right, the `written_bytes` is a delta. See line 555 https://github.com/pingcap/tikv/pull/2701/files#diff-a2b393fbe5f436bb88e05e100af8c1b4L555